src/rapidata/service/credential_manager.py

```diff
@@ -0,0 +1,227 @@
+import json
+import os
+import time
+import webbrowser
+from datetime import datetime, timezone
+from io import TextIOWrapper
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple
+
+import requests
+from pydantic import BaseModel
+
+
+class ClientCredential(BaseModel):
+    display_name: str
+    client_id: str
+    client_secret: str
+    endpoint: str
+    created_at: datetime
+    last_used: datetime
+
+    @classmethod
+    def from_dict(cls, data: Dict):
+        return cls(
+            display_name=data["display_name"],
+            client_id=data["client_id"],
+            client_secret=data["client_secret"],
+            endpoint=data["endpoint"],
+            created_at=datetime.fromisoformat(data["created_at"]),
+            last_used=datetime.fromisoformat(data["last_used"]),
+        )
+
+    def to_dict(self) -> Dict:
+        return {
+            "display_name": self.display_name,
+            "client_id": self.client_id,
+            "client_secret": self.client_secret,
+            "endpoint": self.endpoint,
+            "created_at": self.created_at.isoformat(),
+            "last_used": self.last_used.isoformat(),
+        }
+
+    def get_display_string(self):
+        return f"{self.display_name} - Client ID: {self.client_id} (Created: {self.created_at.strftime('%Y-%m-%d %H:%M:%S')})"
+
+
+class BridgeToken(BaseModel):
+    read_key: str
+    write_key: str
+
+
+class CredentialManager:
+    def __init__(
+        self,
+        endpoint: str,
+        cert_path: str | None = None,
+        poll_timeout: int = 60,
+        poll_interval: int = 1,
+    ):
+        self.endpoint = endpoint
+        self.cert_path = cert_path
+        self.poll_timeout = poll_timeout
+        self.poll_interval = poll_interval
+
+        self.config_dir = Path.home() / ".config" / ".rapidata"
+        self.config_path = self.config_dir / "credentials.json"
+
+        # Ensure config directory exists
+        self.config_dir.mkdir(parents=True, exist_ok=True)
+
+    def _read_credentials(self) -> Dict[str, List[ClientCredential]]:
+        """Read all stored credentials from the config file."""
+        if not self.config_path.exists():
+            return {}
+
+        try:
+            with open(self.config_path, "r") as f:
+                data = json.load(f)
+                return {
+                    env: [ClientCredential.from_dict(cred) for cred in creds]
+                    for env, creds in data.items()
+                }
+        except json.JSONDecodeError:
+            return {}
+
+    def _write_credentials(
+        self, credentials: Dict[str, List[ClientCredential]]
+    ) -> None:
+        data = {
+            env: [cred.to_dict() for cred in creds]
+            for env, creds in credentials.items()
+        }
+
+        with open(self.config_path, "w") as f:  # type: TextIOWrapper
```

Comment:
Pylance gives this error here:
Type annotation not supported for this statementPylancereportInvalidTypeForm

But it doesn't complain if it's simply removed. Is PyCharm complaining here otherwise?